server/src/main/java/org/elasticsearch/tasks/TaskResult.java

```diff
@@ -76,7 +76,7 @@ public TaskResult(TaskInfo task, Exception error) throws IOException {
      * Construct a {@linkplain TaskResult} for a task that completed successfully.
      */
     public TaskResult(TaskInfo task, ToXContent response) throws IOException {
-        this(true, task, null, toXContent(response));
+        this(true, task, null, XContentHelper.toXContent(response, Requests.INDEX_CONTENT_TYPE, true));
```

Comment:
I don't think we store it in the human readable format at the moment. Do we need to make it human readable?

Reply:
Good point. I went with human readable as I think at then these responses will be consumed by users, so it might be best to store the human readable form (if any). But I don't feel strongly about it. Happy to do as you suggest.

Reply:
I am ok with human readable, my only concern is that it is somewhat breaking change since it changes the format (by adding new fields). @nik9000 what do you think about this change?

Reply:
I believe this only changes the stuff in the `task` part. Which *should* be fine because we don't map that anyway. And we kind of expect users to be ok with *new* fields in the http responses and this is fairly similar conceptually. So I'm +1 on it.